packages/backend/server/schema.prisma

```diff
@@ -48,20 +48,53 @@ model User {
   createdAt     DateTime                  @default(now()) @map("created_at") @db.Timestamptz(6)
   /// Not available if user signed up through OAuth providers
   password      String?                   @db.VarChar
-  features      UserFeatureGates[]
+  features      UserFeatures[]
 
   @@map("users")
 }
 
-model UserFeatureGates {
+// feature gates is a way to enable/disable features for a user
+// for example:
+// - early access is a feature that allow some users to access the insider version
+// - pro plan is a quota that allow some users access to more resources after they pay
+model UserFeatures {
+  id        String @id @default(uuid()) @db.VarChar
+  userId    String @map("user_id") @db.VarChar
+  featureId String @db.VarChar
+
+  // we will record the reason why the feature is enabled/disabled
+  // for example:
+  // - pro_plan_v1: "user buy the pro plan"
+  reason    String   @db.VarChar
+  // record the quota enabled time
+  createdAt DateTime @default(now()) @map("created_at") @db.Timestamptz(6)
+  // record the quota expired time, pay plan is a subscription, so it will expired
+  expiresAt DateTime @map("expires_at") @db.Timestamptz(6)
```

Comment:
does it make more sense to use `expiredAt` so we don't need to hard code a far date for free feature?